backend/cfg/vectorize.ml

```diff
@@ -134,7 +138,327 @@ module Dependency_graph = struct
       ~terminator:(fun instruction -> print_node (`Terminator instruction))
 end
 
-let dump ppf cfg_with_layout ~(dependency_graph : Dependency_graph.t) ~msg =
+module Adjacent_memory_accesses = struct
+  module Address = struct
+    type op =
+      | Load
+      | Store
+
+    type t =
+      { op : op;
+        memory_chunk : Cmm.memory_chunk;
+        addressing_mode : Arch.addressing_mode;
+        instruction : Instruction.t
+      }
+
+    let create_loads (basic_instruction : Cfg.basic Cfg.instruction) : t option
+        =
+      let desc = basic_instruction.desc in
+      let instruction = `Basic basic_instruction in
+      match desc with
+      | Op op -> (
+        match op with
+        | Load { memory_chunk; addressing_mode; _ } ->
+          Some { op = Load; memory_chunk; addressing_mode; instruction }
+        | Store _ -> None
+        | Move | Reinterpret_cast _ | Static_cast _ | Spill | Reload
+        | Const_int _ | Const_float32 _ | Const_float _ | Const_symbol _
+        | Const_vec128 _ | Stackoffset _ | Intop _ | Intop_imm _
+        | Intop_atomic _ | Floatop _ | Csel _ | Probe_is_enabled _ | Opaque
+        | Begin_region | End_region | Specific _ | Name_for_debugger _ | Dls_get
+        | Poll | Alloc _ ->
+          None)
+      | Reloadretaddr | Pushtrap _ | Poptrap | Prologue | Stack_check _ -> None
+
+    let create_stores (basic_instruction : Cfg.basic Cfg.instruction) : t option
+        =
+      let desc = basic_instruction.desc in
+      let instruction = `Basic basic_instruction in
+      match desc with
+      | Op op -> (
+        match op with
+        | Load _ -> None
+        | Store (memory_chunk, addressing_mode, _) ->
+          Some { op = Store; memory_chunk; addressing_mode; instruction }
+        | Move | Reinterpret_cast _ | Static_cast _ | Spill | Reload
+        | Const_int _ | Const_float32 _ | Const_float _ | Const_symbol _
+        | Const_vec128 _ | Stackoffset _ | Intop _ | Intop_imm _
+        | Intop_atomic _ | Floatop _ | Csel _ | Probe_is_enabled _ | Opaque
+        | Begin_region | End_region | Specific _ | Name_for_debugger _ | Dls_get
+        | Poll | Alloc _ ->
+          None)
+      | Reloadretaddr | Pushtrap _ | Poptrap | Prologue | Stack_check _ -> None
+
+    let args (t : t) =
+      let args = Instruction.args t.instruction in
+      match t.op with
+      | Load -> args
+      | Store -> Array.sub args 1 (Array.length args - 1)
+
+    let print_memory_chunk ppf (t : t) =
```

Comment:
It looks like a copy from `Printcmm` , can the one in `Printcmm` be used directly.